esphome/components/micro_wake_word/__init__.py

```diff
@@ -319,42 +412,83 @@ async def to_code(config):
     cg.add_build_flag("-DTF_LITE_DISABLE_X86_NEON")
     cg.add_build_flag("-DESP_NN")
 
-    model_config = config.get(CONF_MODEL)
-    data = []
-    if model_config[CONF_TYPE] == TYPE_GIT:
-        # compute path to model file
-        key = f"{model_config[CONF_URL]}@{model_config.get(CONF_REF)}"
-        base_dir = Path(CORE.data_dir) / DOMAIN
-        h = hashlib.new("sha256")
-        h.update(key.encode())
-        file: Path = base_dir / h.hexdigest()[:8] / model_config[CONF_FILE]
+    if on_wake_word_detection_config := config.get(CONF_ON_WAKE_WORD_DETECTED):
+        await automation.build_automation(
+            var.get_wake_word_detected_trigger(),
+            [(cg.std_string, "wake_word")],
+            on_wake_word_detection_config,
+        )
 
-    elif model_config[CONF_TYPE] == TYPE_LOCAL:
-        file = Path(model_config[CONF_PATH])
+    features_step_size = None
 
-    elif model_config[CONF_TYPE] == TYPE_HTTP:
-        file = _compute_local_file_path(model_config) / "manifest.json"
+    if vad_model := config.get(CONF_VAD):
+        cg.add_define("USE_MWW_VAD")
```

Comment:
I know its long, but we need to keep consistency and use the actual component name =)
Also please add it to `esphome/core/defines.h` in the esp32 section so it is set during development and CI.
```suggestion
        cg.add_define("USE_MICRO_WAKE_WORD_VAD")
```

Reply:
I get lazy sometimes :)